w3af/core/data/fuzzer/mutants/tests/test_urlparts_mutant.py

```diff
@@ -129,3 +129,22 @@ def test_valid_results_double_encoding(self):
         generated_urls = set([m.get_url().url_string for m in generated_mutants])
 
         self.assertEqual(set(expected_urls), generated_urls)
+
+    def test_forced_url_parts(self):
```

Comment:
Would like to see a unittest which includes a query string containing the string `foo`. The expected result for that test would be that the QS is not fuzzed.

Reply:
I have added the requested test.

The existing implementation did not handle query strings in the first place, so I added additional behavior to the `get_uri` method (since `get_url` already strips out query strings) that will preserve the query string.